### PR TITLE
refactor(nc-gui): defer loading project to subpage

### PR DESCRIPTION
**Pull request by bcakmakoglu**
## Change Summary

* defer loading to subpage
* unblock smartsheet page and show loading spinner while content is loading
* defer adding tabs until project was loaded

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [ ] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [x] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

Provide summary of changes.

## Additional information / screenshots (optional)

Anything for maintainers to be made aware of

### CODE REVIEW

File: packages/nc-gui/composables/useSharedView.ts [hunk 1]
```diff
@@ -58,7 +60,7 @@ export function useSharedView() {
     const pageSize = paginationData.value.pageSize || 25
 
     const { data } = await $api.public.dataList(
-      sharedView?.value?.uuid,
+      (sharedView.value as any)?.uuid,
```
Comment (pranavxc) [inline packages/nc-gui/composables/useSharedView.ts:63]: Could it be `as ViewType`?

Comment (bcakmakoglu) [reply packages/nc-gui/composables/useSharedView.ts:63]: Nope because ViewType is missing `uuid` otherwise it would work without the type hack :(

Comment (pranavxc) [reply packages/nc-gui/composables/useSharedView.ts:63]: Yes it have `uuid`

```ts
export interface ViewType {
  id?: string;
  title: string;
  deleted?: boolean;
  order?: number;
  fk_model_id?: string;
  slug?: string;
  uuid?: string;
  show_system_fields?: boolean;
  lock_type?: 'collaborative' | 'locked' | 'personal';
  type?: number;
  view?: FormType | GridType | GalleryType;
}
```

Rebuild `nocodb-sdk` and check it out.

File: packages/nc-gui/pages/index/index/index.vue [hunk 2]
```diff
@@ -175,18 +175,18 @@ const getProjectPrimary = (project: ProjectType) => {
     </div>
 
     <TransitionGroup name="layout" mode="out-in">
-      <div v-if="isLoading">
+      <div v-if="isLoading" key="skeleton">
```
Comment (pranavxc) [inline packages/nc-gui/pages/index/index/index.vue:178]: Why there is a static key for these elements and any specific reason to keep them?

**Review by pranavxc — CHANGES_REQUESTED** [review]

File: packages/nc-gui/pages/index/index/index.vue (hunk 2 above)
Comment (bcakmakoglu) [reply packages/nc-gui/pages/index/index/index.vue:178]: Yeah Vue's `transition-group` requires a key on elements :)

File: packages/nc-gui/composables/useSharedView.ts (hunk 1 above)
Comment (bcakmakoglu) [reply packages/nc-gui/composables/useSharedView.ts:63]: Ah okay, I missed the update then, will revert on those changes ^^

**Review by pranavxc — APPROVED** [review]

**Comment by bcakmakoglu** [comment]
@dstala I am not sure if the test fail because of an issue here :s
Could you possibly check if that is the case or maybe even do small manual check if everything seems fine?
On my end everything works well, that's why I am a bit confused

**Review by wingkwong — APPROVED** [review]

**Review by dstala — APPROVED** [review]

**Pull request merged**